### PR TITLE
update geany.glade to target gtk+ 3.20

### LINKED ISSUE #2860: GTK 3-ify the Glade file
When geany.glade is opened in a modern Glade there is a message that it should be checked in Glade 3.8 for deprecated widgets.

Can "somebody" who has Glade 3.8 please check it and post the results so it can be upgraded now GTK 2 is no longer supported.

### PR DESCRIPTION
**Pull request by xiota**
This PR updates `geany.glade` to target gtk+ ~3.0~ 3.20.  Resolves #2860.

* Glade 3.8.6 reports: Project geany.glade has no deprecated widgets or version mismatches.
* Used Glade 3.38.2 to resave geany.glade with target set to gtk+ ~3.0~ 3.20
* ~Removed the following properties to prevent warnings:~
  + ~Property 'Focus on click' of object class 'Widget' was introduced in gtk+ 3.20~
    - ~all were set to False.~
  + ~Property 'X align' of object class 'Label' was introduced in gtk+ 3.16~
    - ~all were set to 0 or 1.~
  + ~Property 'Y align' of object class 'Label' was introduced in gtk+ 3.16~
    - ~only one instance, was set to 0.~

Here is a list of the deprecated items (specific instances removed to keep the list manageable):

* Object class 'Alignment' from gtk+ 3.20 is deprecated
* Object class 'Horizontal Box' from gtk+ 3.20 is deprecated
* Object class 'Horizontal Panes' from gtk+ 3.20 is deprecated
* Object class 'Image Menu Item' from gtk+ 3.20 is deprecated
* Object class 'Table' from gtk+ 3.20 is deprecated
* Object class 'Vertical Box' from gtk+ 3.20 is deprecated
* Object class 'Vertical Panes' from gtk+ 3.20 is deprecated
* Property 'Current Color' of object class 'Color Button' is deprecated
* Property 'Horizontal alignment for child' of object class 'Button' is deprecated
* Property 'Rules Hint' of object class 'Tree View' is deprecated
* Property 'Stock ID' of object class 'Image' is deprecated
* Property 'Use stock' of object class 'Button' is deprecated

### CODE REVIEW

**Comment by elextr** [comment]
I havn't had time to get back to #2860 so thanks for this.

When you say "removed properties" did Glade remove them or you remove them?

I am not sure what is the earliest GTK Geany targets, but GTK3.20 was early 2016, so nearly six years, which IMHO is reasonable enough since most major distros only provide 5 year LTS AFAIK.  If someone needs Geany to support an older version of GTK they can run a branch.  The last thing we want is to return to the situation where contributors have to use old versions of Glade that their distros don't provide (although it did reduce the number of glade file changes :)

**Comment by xiota** [comment]
> I havn't had time to get back to #2860 so thanks for this.

Glad to do this.  Figured there's just too much for you to do.

> When you say "removed properties" did Glade remove them or you remove them?

I removed them and resaved.  The files saved by Glade 3.38.2 appear to be stable.

> I am not sure what is the earliest GTK Geany targets, but GTK3.20 was early 2016, so nearly six years, which IMHO is reasonable enough since most major distros only provide 5 year LTS AFAIK. If someone needs Geany to support an older version of GTK they can run a branch. The last thing we want is to return to the situation where contributors have to use old versions of Glade that their distros don't provide (although it did reduce the number of glade file changes :)

The problem is ... Targetting lower than 3.20 produces those "introduced in" messages.  Targetting higher than 3.0 produces deprecated warnings.  I didn't know the correct approach, so just picked one and wait for feedback.

I just found some visual differences, and it does look better with the properties set.  I'll put them back and target gtk+3.20.

**Comment by xiota** [comment]
Here is a list of the deprecated items (specific instances removed to keep the list manageable):

* Object class 'Alignment' from gtk+ 3.20 is deprecated
* Object class 'Horizontal Box' from gtk+ 3.20 is deprecated
* Object class 'Horizontal Panes' from gtk+ 3.20 is deprecated
* Object class 'Image Menu Item' from gtk+ 3.20 is deprecated
* Object class 'Table' from gtk+ 3.20 is deprecated
* Object class 'Vertical Box' from gtk+ 3.20 is deprecated
* Object class 'Vertical Panes' from gtk+ 3.20 is deprecated
* Property 'Current Color' of object class 'Color Button' is deprecated
* Property 'Horizontal alignment for child' of object class 'Button' is deprecated
* Property 'Rules Hint' of object class 'Tree View' is deprecated
* Property 'Stock ID' of object class 'Image' is deprecated
* Property 'Use stock' of object class 'Button' is deprecated

**Comment by elextr** [comment]
TL;DR don't worry about deprecated here, thats another problem

Yeah, a lot of things are deprecated between GTK3 and GTK4.  I don't know much about GTK4 but those that do have said it may be difficult to reproduce some of the GTK3 functionality our current UI relies on in GTK4.  We need somebody who really knows about GTK3 and GTK4 to identify those so we can consider alternative UI operation.  You can turn deprecations on when building Geany to see how much there is, we had to turn them off because they were hiding real warnings.  So these should not be an issue.

Yeah, I believe 3.20 is ok to target, if anyone disagrees shout, happy to make you a branch to maintain older support. (If it was totally up to me I would say 3.24, but I know there are others using olde stable distros, so 3.20 is a reasonable compromise IMO).

**Comment by xiota** [comment]
> don't worry about deprecated here, thats another problem

Sounds good.

> ... 3.20 is ok to target... If it was totally up to me I would say 3.24, but I know there are others using olde stable distros...

Looks like both Ubuntu 18.04 (EOL in 2023) and Debian stretch are using GTK 3.22.  If someone really needs something from a newer (than 3.20) GTK release, it's easy enough to change the number.

> You can turn deprecations on when building Geany to see how much there is, we had to turn them off because they were hiding real warnings.

Maybe I'll take a look at the deprecation warnings after #3019.

**Comment by kugel-** [comment]
3.20 sounds good

**Comment by xiota** [comment]
MinGW32 CI is using GTK 3.8.2.

**Comment by kugel-** [comment]
I was just about to do the same when I rediscovered this PR. I think by now we're golden even for 3.22, but 3.20 is a good step for now and soon we'll just only support the last gtk3 release (3.24), at some point after the Ubuntu 22.04 LTS release.

Merging this..

**Comment by kugel-** [comment]
Merged https://github.com/geany/geany/commit/305b324a1b5c29dca5184d6917011b7067870c3a
